client/layout/guided-tours/README.md

```diff
@@ -3,4 +3,113 @@ Guided Tours
+
+All of these different ways to leverage the action log constitute *triggers**. Presently, we have one family of triggers: triggers based on navigation to a specific path. This is what is at play when tour authors define a tour with `<Tour path="/design">`, for instance.
+
+The last high-level piece to understand is that Guided Tours barely has any explicit state (e.g. "which step of a tour are we in?", "how long is the tour taking to complete?"), instead relying as much as possible on the action log, which includes a dedicated `GUIDED_TOUR_UPDATE` action to signal tour transitions. The result is that, instead of a hypothetical `state.ui.guidedTours.isTourRunning && <Tour />`, we find a cascade of selectors that ultimately compute the state of Guided Tours. This is achievable thanks to a lot of memoization with `createSelector`:
+
+```text
+getGuidedTourState
+└── findEligibleTour
+    ├── findOngoingTour
+    └── findTriggeredTour
+        └── getToursFromFeaturesReached
+```
+
+#### The selectors
+
+Circling back to Guided Tours — there is one external selector: `getGuidedTourState`. It essentially wraps around [`findEligibleTour`][findEligibleTour], which is where an outline of GT's decision algorithm becomes visible:
+
+```js
+export const findEligibleTour = createSelector(
+	state => {
+		if ( shouldBailAllTours( state ) ) {
+			return;
+		}
+
+		return findOngoingTour( state ) ||
+			! shouldBailNewTours( state ) && (
+				findRequestedTour( state ) ||
+				findTriggeredTour( state )
+			) || undefined;
+	},
+	// Though other state selectors are used in `findEligibleTour`'s body,
+	// we're intentionally reducing the list of dependants to the following:
+	[ getActionLog, getToursHistory ]
+);
+```
+
+Let's break this down:
+
+- from the last line, the primary sources of information are `actionLog` and the tours "history", which is a user preference in `state.preferences`;
+- save for some bailing mechanisms, the decision-making process will favor _ongoing_ tours, then tours that have been explicitly _requested_ (i.e., via query arguments like `?tour=tourName`), then default to tours that can be _triggered_ based on actions that have been tracked in `actionLog`;
+- if nothing is found, `undefined` means that no tour is to be selected.
+
+`findOngoingTour` and `findRequestedTour` are the least interesting. `findTriggeredTour` is the extensible selector. As of this writing, this is what it looks like:
+
+```js
+const findTriggeredTour = state => {
+	const toursFromTriggers = uniq( [
+		...getToursFromFeaturesReached( state ),
+		// Right now, only one source from which to derive tours, but we may
+		// have more later. Examples:
+		// ...getToursFromPurchases( state ),
+		// ...getToursFromFirstActions( state ),
+	] );
+
+	const toursToDismiss = uniq( [
+		// Same idea here.
+		...getToursSeen( state ),
```

Comment:
I think we should the "tours show up again and again because the prefs syncing fails silently because expired 2fa token" gotcha here (or somewhere else).